homeassistant/components/leneda/__init__.py

```diff
@@ -0,0 +1,84 @@
+"""The Leneda integration."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.device_registry import DeviceEntry
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_ENERGY_ID, CONF_METERING_POINTS, DOMAIN
+from .coordinator import LenedaCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+type LenedaConfigEntry = ConfigEntry[LenedaCoordinator]
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Leneda component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: LenedaConfigEntry) -> bool:
+    """Set up Leneda from a config entry."""
+    coordinator = LenedaCoordinator(hass, entry)
+    await coordinator.async_config_entry_first_refresh()
+    entry.runtime_data = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id, None)
```

Comment:
As you are using `runtime_data`, this could be simplified, example:

https://github.com/home-assistant/core/blob/6d809b0b5ad5993a973c21122f4da713645b96e3/homeassistant/components/aosmith/__init__.py#L62

Reply:
Thank you, done.